lib/vm/src/instance/mod.rs

```diff
@@ -807,8 +712,10 @@ impl Instance {
         len: u32,
     ) -> Result<(), Trap> {
         // https://webassembly.github.io/reference-types/core/exec/instructions.html#exec-memory-copy
-
-        let memory = self.memory(memory_index);
+        let memory = unsafe {
+            // TODO: safety comment
```

Comment:
Safety comment here.

Reply:
This is a refactor of its own – a lot of functions on instance don't correctly specify whether they are safe or not. Most notably most of the memory-related methods require the caller to synchronize, but eventually this requirement gets lost in one of the `unsafe` blocks here instead of being propagated to the user.

I don't want to deal with this here, especially since its a preexisting condition.